### PR TITLE
Add server skeleton

### PR DESCRIPTION
**Pull request by VolodymyrFihurniak**

### CODE REVIEW

File: server/src/business/authWorker.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      } catch (err) {
+        serverLogger.error("Error deleting token");
+      } finally {
+        serverLogger.trace("AuthWorker.refreshToken completed");
+        this.db.closeConnection();
+      }
+      serverLogger.error("Refresh token expired");
+      return null;
+    }
+    const token = await this.db.findTokenByRefreshToken(refreshTokenId);
```
Comment (denbykov) [inline server/src/business/authWorker.ts:93]: Rename it to the "findAccessTokenByRefreshTokenId"

File: server/src/business/authWorker.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    } catch (err) {
+      serverLogger.error("Error updating token");
+      throw new Error("Error updating token");
+    } finally {
+      serverLogger.trace("AuthWorker.refreshToken completed");
+      this.db.closeConnection();
+    }
+    return {
+      accessToken: newToken,
+    };
```
Comment (denbykov) [inline server/src/business/authWorker.ts:119]: I think you can create a separate function that will create a token, update database and return it

File: server/src/business/authWorker.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      return null;
+    }
+    if (!this.verifyExpiredToken(refreshToken, "refresh")) {
+      try {
+        await this.db.deleteToken(refreshToken, "refresh");
+      } catch (err) {
+        serverLogger.error("Error deleting token");
+      } finally {
+        serverLogger.trace("AuthWorker.refreshToken completed");
+        this.db.closeConnection();
```
Comment (denbykov) [inline server/src/business/authWorker.ts:88]: Why are you closing a db connection here?

File: server/src/controllers/auth.controller.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        message: "Login failed",
+      });
+    }
+
+    return res.status(200).send({
+      tokens,
+    });
+  };
+
+  public refreshToken = async (req: Request, res: Response) => {
```
Comment (denbykov) [inline server/src/controllers/auth.controller.ts:33]: I believe that you are creating an access token here, so name it accessToken

File: server/src/controllers/base.controller.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Config } from "@src/entities/config";
+
+class BaseController {
+  protected apiURI: string;
+  protected apiURIAuth: string;
+  protected _config: Config;
```
Comment (denbykov) [inline server/src/controllers/base.controller.ts:6]: Why do you have an underscore in the name of this property here?

File: server/src/middlewares/auth0Middleware.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  return async (
+    req: Express.Request,
+    res: Express.Response,
+    next: Express.NextFunction
+  ) => {
+    const authWorker = new AuthWorker(await new Database(config), config);
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(403).send({
+        message: "Token invalid",
```
Comment (denbykov) [inline server/src/middlewares/auth0Middleware.ts:16]: "Token is invalid" would be better

File: server/src/middlewares/auth0Middleware.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        _error: 1,
+      });
+    }
+    const token = authHeader.split(" ")[1];
+    const auth = await authWorker.auth(token);
+    if (!auth) {
+      return res.status(403).send({
+        message: "Token invalid",
```
Comment (denbykov) [inline server/src/middlewares/auth0Middleware.ts:24]: "Token is invalid" would be better

File: server/src/middlewares/index.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+import { requestLogger } from "./loggerRequest";
```
Comment (denbykov) [inline server/src/middlewares/index.ts:1]: Weird naming

File: server/src/server.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+const env = dotenv.config({ path: "config/.env" }).parsed || {};
+const configJson = parseConfigJSON(
+  JSON.parse(fs.readFileSync("config/config.json", "utf-8"))
+);
+const config = new Config(env, configJson);
+const app = new App(config);
+
+if (config.appPort == undefined || config.appHost == undefined) {
+  serverLogger.error("APP_PORT and APP_HOST are not defined");
```
Comment (denbykov) [inline server/src/server.ts:17]: These variables have different names now

File: server/src/data/Database.ts [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    port: parseInt(config.dbPort),
+    password: config.dbPasswrd,
+    max: parseInt(config.dbMax),
+  };
+  return _config;
+};
+
+export class Database implements IDatabase {
+  public config: Database.DBClientConfig;
+  public client: Promise<pg.PoolClient>;
```
Comment (denbykov) [inline server/src/data/Database.ts:21]: PoolClient is a Pool manager or an object being managed by the pool? The second seems to be true, and if that is the case, then it has to be rewritten

**Review by denbykov — CHANGES_REQUESTED** [review]

File: server/config/log4j.properties [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+    "appenders": {
+        "out": { "type": "stdout" },
+        "app": { "type": "file", "filename": "logs/app.log" }
+    },
+    "categories": {
+        "default": { "appenders": ["out", "app"], "level": "debug", "enableCallStack": true },
+        "server": { "appenders": ["out", "app"], "level": "debug", "enableCallStack": true }
```
[the remark this answers is not in the document]
Comment (VolodymyrFihurniak) [reply server/config/log4j.properties]: Fixed

**Review by denbykov — APPROVED** [review]

**Pull request merged**